File: CourseRegistrationSystem/src/Student.java
import java.util.ArrayList;
import java.util.List;

class Student extends User {

    private String studentId;
    private List<Course> enrolledCourses;
    private List<Course> requestedCourses;
    private Transcript transcript;

    public Student(String name, String surname, String username, String password, String studentId,
            List<Course> enrolledCourses, List<Course> requestedCourses, Transcript transcript) {
        super(username, surname, name, password);
        this.studentId = studentId;
        this.enrolledCourses = enrolledCourses;
        this.requestedCourses = requestedCourses;
        this.transcript = transcript;
    }

    public Transcript viewTranscript() {

        return this.transcript;
    }

    public void clearRequestedCourses() {
        this.requestedCourses.clear();
        // JSONDA REQUEST SİLİNECEK
    }

    public String getStudentId() {
        return studentId;
    }

    public List<Course> getEnrolledCourses() {
        return enrolledCourses;
    }

    public List<Course> getRequestedCourses() {
        return requestedCourses;
    }
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Name: ").append(getName()).append("\n");
        sb.append("Surname: ").append(getSurname()).append("\n");
        sb.append("Student ID: ").append(getStudentId()).append("\n");
        return sb.toString();
    }

}

File: iteration3/Student.py
class Student:
    total_credit = 0
    def __init__(self, username, name, surname, password, studentID, GPA, StSemester, enrolledCourses, requestedCourses, projectAssistant, transcript):
        self.username=username
        self.name=name
        self.surname=surname
        self.password=password
        self.studentID=studentID
        self.GPA=GPA
        self.StSemester=StSemester
        self.enrolledCourses=enrolledCourses
        self.requestedCourses=requestedCourses
        self.projectAssistant=projectAssistant
        self.transcript=transcript
    def getUsername(self):
        return self.username
    def getName(self):
        return self.name
    def getSurname(self):
        return self.surname
    def getPassword(self):
        return self.password
    def getstudentID(self):
        return self.studentID
    def getGPA(self):
        return self.GPA
    def getStSemester(self):
        return self.StSemester
    def getEnrolledCourses(self):
        return self.enrolledCourses
    def getRequestedCourses(self):
        return self.requestedCourses
    def getProjectAssistant(self):
        return self.projectAssistant
    def getTranscript(self):
        return self.transcript
    def calculateTotalCredit(self):
        global total_credit
        for course in self.getEnrolledCourses:
            total_credit += course.credit
        return total_credit
    def getTotalCredit():
        global total_credit
        return total_credit
        